src/util/make-knex.js

```diff
@@ -10,24 +10,23 @@ import batchInsert from './batchInsert';
 export default function makeKnex(client) {
   // The object we're potentially using to kick off an initial chain.
   function knex(tableName, options) {
```

Comment:
There are more changes here than I am comfortable with, TBH. Main reason - there was actually a creeping bug after withUserParams change - invoking knex('tableName') would silently create a builder from wrong (original) knex instance. While figuring out a solution I've implemented a stricter separation between knex() function (which should be as lightweight as possible) and the actual knex execution context (that knex function proxies to). I'm still not happy about our magical knex entity which is function, object and class instance at the same time, but at least there is clearer separation of what is responsible for what now)